test/mod/test_util_jsoncomm.py

```diff
@@ -165,3 +166,52 @@ def echo(socket):
 
         msg = self.client.recv()
         assert msg[0] == {}
+
+    def test_accept_race(self):
```

Comment:
Ref. my previous comment on this topic. Why don't we flip it around, never block, and never rely on there being a socket when we call `accept()`, then it doesn't matter?

Reply:
This is not easy for me. I like tests that verify kernel invariants! And I like relying on them! But I think in this case I am with Tom, because there is just too many moving parts that we have to rely on to not change.

1) I wasn't even aware that the kernel does not remove `SOCK_DEAD` sockets from the receive queue. This is at least something we could reasonably well test for, just as you did.
2) You rely on `EPOLLIN` only being raised for incoming connections. Again, this I would also be fine with on listener sockets on **local** connections. But I am even less sure this might not change in the future.
3) You rely on the event-loop only waking you up for `EPOLLIN`. This I am pretty sure is **not** true. I checked the epoll-cpython selector [1], and it wakes readers if **any flag other than EPOLLOUT** is set. So you do get woken up for **EPOLLERR**, **EPOLLPRI**, **EPOLLHUP**, etc., and for sockets there can always be asynchronous errors in the off-channel error-queue.

So, I have to admit, I think we have to take Tom's advice and assume `accept()` can fail. As consolation, I would be fine merging it spelled as `accpet()`?

[1] https://github.com/python/cpython/blob/bcd47837a9bf4806e559b40df73869493efcce27/Lib/selectors.py#L474

Reply:
I did flit it around, never block but then I dropped this test because I could not find a way to actually trigger `accept` **NOT** returning a socket.